src/modules/site-settings/edit-site-details.tsx

```diff
@@ -73,12 +74,13 @@ export default function EditSiteDetails( { currentWpVersion, onSave }: EditSiteD
 		}
 		setIsEditingSite( true );
 		setIsChangeWpError( '' );
-		try {
-			const running = selectedSite.running;
 
-			const hasWpVersionChanged = selectedWpVersion !== currentWpVersion;
-			const hasPhpVersionChanged = selectedPhpVersion !== selectedSite.phpVersion;
-			const needsRestart = running && ( hasWpVersionChanged || hasPhpVersionChanged );
+		const hasWpVersionChanged = selectedWpVersion !== currentWpVersion;
+		const hasPhpVersionChanged = selectedPhpVersion !== selectedSite.phpVersion;
+		const needsRestart = selectedSite.running && ( hasWpVersionChanged || hasPhpVersionChanged );
+		setNeedsRestart( needsRestart );
```

Comment:
Nit, but I would propose this way, as, IMO, more simpler to read and understand at a glance
Also we don't trigger setting state every time, even when it's definitely not necessary
```suggestion
		if ( selectedSite.running ) {
			const hasWpVersionChanged = selectedWpVersion !== currentWpVersion;
			const hasPhpVersionChanged = selectedPhpVersion !== selectedSite.phpVersion;
			
			setNeedsRestart( hasWpVersionChanged || hasPhpVersionChanged );
		}
```